student-work/yerim/week_1/working_with_classes.py

```diff
@@ -1,15 +1,63 @@
 
-class Myclass:
-    pass
+class BaseClass:
 
+    def __init__(self, first_name, last_name, email, question):
+        self.first_name = first_name
+        self.last_name = last_name
+        self.email = email
+        self.question = question
 
+    def ask_a_question(self):
+        return(self.question)
 
 
+class Student(BaseClass):
 
+    def __init__(self, first_name, last_name, email, question, books_add, books_remove):
+        super().__init__(first_name, last_name, email, question)
+        self.books_dict = {}
+        self.books_add = books_add
+        self.books_remove = books_remove
 
+    def add_book(self):
+        for book in self.books_add:
+            if book in self.books_dict.keys():
+                self.books_dict[book] =+ 1
+            else:
+                self.books_dict[book] = 1
+        return(self.books_dict)
 
+    def remove_book(self):
+        for book in self.books_remove:
+            if self.books_dict[book] > 1:
+                self.books_dict[book] -= 1
+            else:
+                del self.books_dict[book]
+        return(self.books_dict)
 
+    def show_books(self):
+        return(self.books_dict.keys())
 
-if __name__ =='__main__':
-    class_inst = Myclass()
-    print(class_inst)
+
+class Teacher(BaseClass):
+    def __init__(self, first_name, last_name, email, question, student_list, size):
+        super().__init__(first_name, last_name, email, question)
+        self.student_list = student_list
+        self.size = size
+
+    def shuffle_class(self):
+
+        from group_challenge.names_challenge import names_func
```

Comment:
Great job! One suggestion would be to move your import statement to the top of the file, usually imports are not made inside methods.